frameworks/file-analysis.rst

```diff
@@ -126,38 +161,77 @@ calculate the MD5 of plain text files:
    new file, FakNcS1Jfe01uljb3
    file_hash, FakNcS1Jfe01uljb3, md5, 397168fd09991a0e712254df7bc639ac
 
-Some file analyzers have tunable parameters that need to be
-specified in the call to :zeek:see:`Files::add_analyzer`:
+File extraction
+^^^^^^^^^^^^^^^
+
+The following example sets up simplistic extraction of observed files to
+disk:
 
 .. code-block:: zeek
 
+    global idx: count = 0;
+
     event file_new(f: fa_file)
         {
         Files::add_analyzer(f, Files::ANALYZER_EXTRACT,
-                            [$extract_filename="myfile"]);
+                            [$extract_filename=fmt("file-%04d", ++idx)]);
         }
 
-In this case, the file extraction analyzer doesn't generate any further
-events, but does have the effect of writing out the file contents to the
-local file system at the location resulting from the concatenation of
-the path specified by :zeek:see:`FileExtract::prefix` and the string,
-``myfile``.  Of course, for a network with more than a single file being
-transferred, it's probably preferable to specify a different extraction
-path for each file, unlike this example.
+The file extraction analyzer now writes the content of each observed file to a
+separate file on disk. The output file name results from concatenating the
+:zeek:see:`FileExtract::prefix` (normally ``./extract_files/``) and the
+enumerated ``file-NNNN`` strings.
+
+In a production setting you'll likely want to include additional information in
+the output, for example from state attached to the provided file record. The
+Zeek distribution ships with a starting point for such approaches: the
+``policy/frameworks/files/extract-all-files.zeek`` script. For additional
+configurability, take a look at the `file-extraction
+<https://github.com/hosom/file-extraction>`_ Zeek package.
+
+Content analysis
```

Comment:
Seems Script-Level would be good in the heading? Maybe "Script-Layer File Data Access" as the first sentence indicates?

Reply:
Good one, yeah.